src/sentry/runner/initializer.py

```diff
@@ -79,6 +79,9 @@ def get_asset_version(settings):
     'mail.use-tls': 'EMAIL_USE_TLS',
     'mail.from': 'SERVER_EMAIL',
     'mail.subject-prefix': 'EMAIL_SUBJECT_PREFIX',
+    'mail.enable-replies': 'SENTRY_ENABLE_EMAIL_REPLIES',
+    'mail.reply-hostname': 'SENTRY_SMTP_HOSTNAME',
+    'mail.mailgun-api-key': 'MAILGUN_API_KEY',
```

Comment:
Do need to be written back to settings? Or can this go in `apply_legacy_settings`? It looks like you added tests to check that they were written back, but I'm curious why that is necessary.